parser/parser_test.go

```diff
@@ -1499,6 +1500,11 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"ALTER TABLE t ENGINE = ''", true},
 		{"ALTER TABLE t ENGINE = 'innodb'", true},
 		{"ALTER TABLE t ENGINE = innodb", true},
+		{"ALTER TABLE t ADD COLUMN a SMALLINT UNSIGNED, ADD COLUMN a SMALLINT", true},
+		{"ALTER TABLE t ADD COLUMN a SMALLINT, ENGINE = '', default COLLATE = utf8_general_ci", true},
+		{"ALTER TABLE t ENGINE = '', COMMENT='', default COLLATE = utf8_general_ci", true},
+		{"ALTER TABLE t ENGINE = '', ADD COLUMN a SMALLINT", true},
+		{"ALTER TABLE t default COLLATE = utf8_general_ci, ENGINE = '', ADD COLUMN a SMALLINT", true},
 
```

Comment:
How about adding a case  ALTER TABLE `foo`.`bar` ENGINE=``     (engine value in back-quote)

Reply:
Done.